samples/dnn/optical_flow.py

```diff
@@ -75,7 +76,25 @@ def motion_to_color(self, flow):
     cv.namedWindow(winName, cv.WINDOW_NORMAL)
     cap = cv.VideoCapture(args.input if args.input else 0)
     hasFrame, first_frame = cap.read()
-    opt_flow = OpticalFlow(args.proto, args.model, args.height, args.width)
+
+    divisor = 64.
+    var = {}
+    var['ADAPTED_WIDTH'] = int(np.ceil(args.width/divisor) * divisor)
+    var['ADAPTED_HEIGHT'] = int(np.ceil(args.height/divisor) * divisor)
+    var['SCALE_WIDTH'] = args.width / float(var['ADAPTED_WIDTH'])
+    var['SCALE_HEIGHT'] = args.height / float(var['ADAPTED_HEIGHT'])
+
+    tmp = tempfile.NamedTemporaryFile(mode='w', delete=True)
```

Comment:
My last suggestion try using https://github.com/opencv/opencv/blob/master/modules/dnn/src/caffe/caffe_importer.cpp#L584 without creating temporary file. 
To do it we need using `self.net = cv.dnn.readNetFromCaffe(proto, model)` instead of `cv.dnn.readNet`

And read weights from caffemodel to bytearray.

```python
    config = ''
    proto = open(args.proto).readlines()
    for line in proto:
        for key, value in var.items():
            tag = "$%s$" % key
            line = line.replace(tag, str(value))
        config += line

    caffemodel = open(args.model, "rb").read()
    opt_flow = OpticalFlow(bytearray(config.encode()), caffemodel, var['ADAPTED_HEIGHT'], var['ADAPTED_WIDTH'])

```